### PR TITLE
HPCC-23646 Use default storageclass for persistent storage volumes

### PR DESCRIPTION
**Pull request by richardkchapman**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [x] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Minikube

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith Please review

File: dockerfiles/hpcc/templates/_util.tpl [hunk 1]
```diff
@@ -58,14 +90,12 @@ data:
...
-volumes:
-{{ include "hpcc.utils.addConfigVolume" . }}
 - name: dllserver-pv-storage
   persistentVolumeClaim:
-    claimName: dllserver-pv-claim
+    claimName: {{ .Values.global.dllserver.existingClaim | default (printf "%s-dllserver-pv-claim" (include "hpcc.utils.fullname" .)) }}
 - name: datastorage-pv
   persistentVolumeClaim:
-    claimName: datastorage-pv-claim
+    claimName: {{ .Values.global.dllserver.existingClaim | default (printf "%s-datastorage-pv-claim" (include "hpcc.utils.fullname" .)) }}
```
Comment (jakesmith) [inline dockerfiles/hpcc/templates/_util.tpl]: should be claimName: {{ .Values.global.dataStorage.existingClaim ...

File: dockerfiles/hpcc/templates/datastorage.yaml [hunk 2]
```diff
@@ -1,28 +1,17 @@
-apiVersion: v1
```
Comment (jakesmith) [inline dockerfiles/hpcc/templates/datastorage.yaml]: we should rename datastorage.yaml to datastorage-claim.yaml (to be consistent with dllserver-claim.yaml)

File: dockerfiles/hpcc/values.yaml [hunk 3]
```diff
@@ -19,13 +32,19 @@ global:
...
+  ## If defined, storageClassName: <storageClass>
+  ## If set to "-", storageClassName: "", which disables dynamic provisioning
+  ## If undefined (the default) or set to null, no storageClassName spec is
+  ##   set, choosing the default provisioner.  (gp2 on AWS, standard on
+  ##   GKE, AWS & OpenStack)
+  ##
+  storageClass: ""
+  accessMode: ReadWriteOnce
+  size: 10Gi
+  # existingClaim: ""
```
Comment (jakesmith) [inline dockerfiles/hpcc/values.yaml]: above 'persistence:' section to be deleted

File: dockerfiles/hpcc/templates/datastorage.yaml (hunk 2 above)
Comment (richardkchapman) [reply dockerfiles/hpcc/templates/datastorage.yaml]: we should rename them both - I have a separate Jira https://track.hpccsystems.com/browse/HPCC-23649 in progress where I intend to address that.

**Review by jakesmith — APPROVED** [review]
@richardkchapman looks good

**Pull request merged**